src/components/SignUpDialogueBox.tsx

```diff
@@ -1,24 +1,89 @@
 // React imports
-import React, { useState } from "react";
+import React, { useState, SyntheticEvent } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useMutation, useApolloClient } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
 
-import UploadFile from "./UploadFile";
+import UploadFile from './UploadFile';
 
 // Type definitions
 type DialogueProps = {
   isVolunteer: boolean;
   setIsVolunteer: any;
 };
 
+const REGISTER_VOLUNTEER = gql`
+  mutation registerVolunteer(
+    $username: String!
+    $password: String!
+    $confirmPassword: String!
+  ) {
+    registerVolunteer(
+      username: $username
+      password: $password
+      confirmPassword: $confirmPassword
```

Comment:
I think for password confirmation, I would do that check client side.  If someone is signing in through the API they can accidentally make a typo in their password :-)